### ISSUE TITLE
Inline all of the sdefaults scripts

### ISSUE THREAD

**Issue by ianthehenry** [ianthehenry/sd#1]
Ported from https://github.com/ianthehenry/dotfiles/issues/1

It would be a little cleaner if `sd` just had the "default" commands as first-class features instead of separate scripts, and it would make installation much easier.

**Comment by ianthehenry** [comment]
@gerardbosch I think that this should work now. I've changed the `sd` interface a bit, so that the `sdefaults/` commands are now optional, which means that it can be "installed" by sourcing a single file in a way that I believe will be compatible with antigen. I haven't actually tested it, as I do not have antigen installed, but if it doesn't work I expect it will take very little work to get it there.

But basically now:

    $ sd help command

Becomes:

    $ sd command --help

The previous `sd help` command is still in `sdefaults/` and you can install it if you want, but `sd` no longer relies on specific scripts in your `~/sd`.

`sd` can still be used as a regular executable as well. The installation instructions in the readme describe the two approaches in slightly more detail.

---

Here's a better way to phrase what I was trying to say before:

I think the fact that `sd` happens to be a shell script is an implementation detail.

I could imagine a future where I rewrite `sd` in C or something, and distribute it as a binary rather than a script, which would break the antigen workflow. But I don't really see myself doing that for the foreseeable future, so this is more an idle musing. It would probably be a lot more work than it's worth, and this workflow is an argument against making that change.

**Comment by ianthehenry** [comment]
One more thing: I've tried to make it work in `bash` and `zsh` identically, so that the function form behaves the same as the executable form. But writing portable shell is very difficult, and it's possible that I've made a mistake. If you encounter any bugs using it as a zsh function, I'd like to hear about them.

Also, note that since the script now runs in the same context as your interactive shell, it's possible there will be problems due to zsh customization -- I won't try to make it work with every combination of zsh globbing option or whatever. Probably the biggest downside of the function approach is that lack of hygiene. Still, it'd be good to hear about them in case there's an easy fix.

**Comment by gerardbosch** [comment]
:tada: That's cool! I've just tested it, I just added to my `.zshrc`:

```console
antigen bundle ianthehenry/sd
```

and BOOM! there `sd` goes!

I've found a very little warning when running `--help` straight on `sd`, so I'm publishing this: https://github.com/ianthehenry/sd/pull/2

I also took the liberty to expand on 'install via plugin manager' instructions here :) https://github.com/ianthehenry/sd/pull/3
